docs/java-testing-and-debugging.md

```diff
@@ -0,0 +1,123 @@
+---
+id: java-testing-and-debugging
+title: Testing and Debugging
+sidebar_label: Testing and Debugging
+---
+
+The Temporal Java SDK provide test framework support to facilitate Workflow unit and integration testing.
+The test framework provides a`TestWorkflowEnvironment` class which includes an in-memory implementation
+of the Temporal service that supports automatic time skipping. This allows you to 
+easily test long-running workflows in seconds, without having to change your workflow code.
+
+You can use the provided `TestWorkflowEnvironment` with an Java unit testing framework on your choice,
+such as JUnit.
+
+The following code implements an unit tests for the `HelloActivity` sample:
+
+```java
```

Comment:
Can we snipsync these examples?
Looks like they come from the samples-java repo?

Reply:
sure. i would rather do it across the board (all sdks) rather than just here now. if you are ok with it i would do a separate PR for that